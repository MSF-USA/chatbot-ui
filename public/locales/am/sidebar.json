{
  "Settings": "ቅንብሮች",
  "New Conversation": "አዲስ ውይይት",
  "New chat": "ዝማኔ አዲስ",
  "Save": "አስቀምጥ",
  "Search conversations...": "ውይይቶችን ፈልግ...",
  "New prompt": "አዲስ ጥያቄ",
  "Dark mode": "ጨለማ ሁነታ",
  "Light mode": "ብርሃን ሁነታ"
}