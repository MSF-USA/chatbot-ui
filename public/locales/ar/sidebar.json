{
  "New folder": "مجلد جديد",
  "New chat": "محادثة جديدة",
  "No conversations.": "لا يوجد محادثات",
  "Search conversations...": "...البحث عن المحادثات",
  "OpenAI API Key": " (أوبن أيه أي) OpenAI API Key (مفتاح واجهة برمجة تطبيقات)",
  "Import data": "استيراد المحادثات",
  "Are you sure?": "هل أنت متأكد؟",
  "Clear conversations": "مسح المحادثات",
  "Export data": "تصدير المحادثات",
  "Dark mode": "الوضع الداكن",
  "Light mode": "الوضع الفاتح",
  "New prompt": "مطلب جديد",
  "No prompts.": "لا يوجد مطالبات.",
  "Search prompts...": "...البحث عن مطالبات",
  "Name": "الاسم",
  "Description": "الوصف",
  "A description for your prompt.": "وصف لمطلبك",
  "Prompt": "مطلب",
  "Prompt content. Use {{}} to denote a variable. Ex: {{name}} is a {{adjective}} {{noun}}": "محتوى المطلب. استخدم {{}} للإشارة إلى متغير. مثال: {{الاسم}} هي {{صفة}} {{اسم}}",
  "Save": "حفظ"
}