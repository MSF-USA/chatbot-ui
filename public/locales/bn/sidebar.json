{
  "New folder": "নতুন ফোল্ডার",
  "New chat": "নতুন আড্ডা",
  "No conversations.": "কোনো আলাপচারিতা নেই।",
  "Search conversations...": "আলাপচারিতা খুঁজুন...",
  "OpenAI API Key": "OpenAI API Key",
  "Import data": "আলাপচারিতা ইমপোর্ট",
  "Are you sure?": "আপনি কি নিশ্চিত?",
  "Clear conversations": "কথোপকথন পরিষ্কার করুন",
  "Export data": "আলাপচারিতা এক্সপোর্ট",
  "New prompt": "নতুন prompt",
  "No prompts.": "কোনো prompts নেই।",
  "Search prompts...": "prompts অনুসন্ধান হচ্ছে...",
  "Name": "নাম",
  "Description": "বর্ণনা",
  "A description for your prompt.": "আপনার Prompt জন্য একটি বিবরণ লিখুন.",
  "Prompt": "প্রম্পট",
  "Prompt content. Use {{}} to denote a variable. Ex: {{name}} is a {{adjective}} {{noun}}": "Prompt content. Use {{}} to denote a variable. Ex: {{name}} is a {{adjective}} {{noun}}",
  "Save": "সংরক্ষণ করুন",
  "Plugin Keys": "প্লাগইন কীসমূহ",
  "Light mode": "আলো মোড",
  "New Conversation": "নতুন কথোপকথন",
  "Settings": "সেটিংস",
  "A name for your prompt.": "আপনার প্রম্পটের জন্য একটি নাম।",
  "Dark mode": "ডার্ক মোড"
}
