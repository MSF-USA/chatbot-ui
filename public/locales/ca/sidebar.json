{
  "New folder": "Nova carpeta",
  "New chat": "Nova conversa",
  "No conversations.": "No hi ha converses.",
  "Search conversations...": "Cerca converses...",
  "OpenAI API Key": "Clau d'API d'OpenAI",
  "Import data": "Importar converses",
  "Are you sure?": "Estàs segur?",
  "Clear conversations": "Esborrar converses",
  "Export data": "Exportar converses",
  "Dark mode": "Mode fosc",
  "Light mode": "Mode clar",
  "New prompt": "Nou prompt",
  "No prompts.": "No hi ha prompts.",
  "Search prompts...": "Cerca prompts...",
  "Name": "Nom",
  "Description": "Descripció",
  "A description for your prompt.": "Descripció del teu prompt.",
  "Prompt": "Prompt",
  "Prompt content. Use {{}} to denote a variable. Ex: {{name}} is a {{adjective}} {{noun}}": "Contingut del missatge. Utilitza {{}} per indicar una variable. Ex: {{nom}} és un {{adjectiu}} {{substantiu}}",
  "Save": "Guardar"
}