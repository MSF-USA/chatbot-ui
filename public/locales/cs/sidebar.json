{
  "Settings": "Nastavení",
  "New Conversation": "Nová konverzace",
  "New chat": "Nový chat",
  "Save": "uložit",
  "Search conversations...": "Vyhledat konverzace...",
  "New prompt": "Nová výzva",
  "Dark mode": "tmavý režim",
  "Light mode": "světlý režim"
}
