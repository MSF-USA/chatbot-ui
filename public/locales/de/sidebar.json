{
  "New folder": "Neuer Ordner",
  "New chat": "Neue Konversation",
  "No conversations.": "Keine Konversationen.",
  "Search conversations...": "Konversationen suchen...",
  "OpenAI API Key": "OpenAI API-Schlüssel",
  "Import data": "Konversationen importieren",
  "Are you sure?": "Bist du sicher?",
  "Clear conversations": "Konversationen löschen",
  "Export data": "Konversationen exportieren",
  "Name": "Name",
  "Plugin Keys": "Plugin-Schlüssel",
  "Prompt": "Aufforderung",
  "Light mode": "Heller Modus",
  "New Conversation": "Neues Gespräch",
  "Settings": "Einstellungen",
  "Search prompts...": "Suchen Sie nach Eingabeaufforderungen...",
  "Save": "Speichern",
  "Prompt content. Use {{}} to denote a variable. Ex: {{name}} is a {{adjective}} {{noun}}": "Inhalt des Prompts. Verwenden Sie {{}} zur Kennzeichnung einer Variable. Beispiel: {{name}} ist ein {{adjective}} {{noun}}",
  "New prompt": "Neue Eingabeaufforderung",
  "A description for your prompt.": "Eine Beschreibung für Ihre Eingabeaufforderung.",
  "A name for your prompt.": "Ein Name für Ihren Prompt.",
  "Dark mode": "Dunkelmodus",
  "No prompts.": "Keine Vorgaben.",
  "Description": "Beschreibung"
}