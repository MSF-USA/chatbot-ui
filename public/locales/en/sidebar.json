{
  "Import data": "Restore chat history",
  "Export data": "Backup chat history"
}