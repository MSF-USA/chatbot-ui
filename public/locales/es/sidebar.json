{
  "New folder": "Nueva carpeta",
  "New chat": "Nueva conversación",
  "No conversations.": "No hay conversaciones.",
  "Search conversations...": "Buscar conversaciones...",
  "OpenAI API Key": "Llave de API de OpenAI",
  "Import data": "Importar conversaciones",
  "Are you sure?": "¿Estás seguro?",
  "Clear conversations": "Borrar conversaciones",
  "Export data": "Exportar conversaciones",
  "New prompt": "Nuevo prompt",
  "No prompts.": "No hay prompts.",
  "Search prompts...": "Buscar prompts...",
  "Name": "Nombre",
  "Description": "Descripción",
  "A description for your prompt.": "Descripción de su prompt.",
  "Prompt": "indicación",
  "Prompt content. Use {{}} to denote a variable. Ex: {{name}} is a {{adjective}} {{noun}}": "Contenido del prompt. Utilice {{}} para indicar una variable. Ej: {{nombre}} es un {{adjetivo}} {{nombre}}",
  "Save": "Guardar",
  "Plugin Keys": "Claves de complemento",
  "Light mode": "modo claro",
  "New Conversation": "Nueva conversación",
  "Settings": "Configuración",
  "A name for your prompt.": "Un nombre para tu aviso.",
  "Dark mode": "modo oscuro"
}