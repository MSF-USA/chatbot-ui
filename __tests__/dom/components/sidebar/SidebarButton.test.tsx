import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { describe, expect, it, vi } from 'vitest';
import '@testing-library/jest-dom';

import { SidebarButton } from '@/components/sidebar/SidebarButton';

describe('SidebarButton', () => {
  const mockIcon = <svg data-testid="test-icon">Icon</svg>;

  it('renders button with text', () => {
    const mockOnClick = vi.fn();
    render(<SidebarButton text="Test Button" icon={mockIcon} onClick={mockOnClick} />);

    const buttonText = screen.getByText('Test Button');
    expect(buttonText).toBeInTheDocument();
  });

  it('renders icon', () => {
    const mockOnClick = vi.fn();
    render(<SidebarButton text="Test" icon={mockIcon} onClick={mockOnClick} />);

    const icon = screen.getByTestId('test-icon');
    expect(icon).toBeInTheDocument();
  });

  it('calls onClick when clicked', () => {
    const mockOnClick = vi.fn();
    render(<SidebarButton text="Clickable" icon={mockIcon} onClick={mockOnClick} />);

    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('calls onClick multiple times', () => {
    const mockOnClick = vi.fn();
    render(<SidebarButton text="Multi Click" icon={mockIcon} onClick={mockOnClick} />);

    const button = screen.getByRole('button');
    fireEvent.click(button);
    fireEvent.click(button);
    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalledTimes(3);
  });

  it('has correct styling classes', () => {
    const mockOnClick = vi.fn();
    render(<SidebarButton text="Styled" icon={mockIcon} onClick={mockOnClick} />);

    const button = screen.getByRole('button');
    expect(button).toHaveClass('flex');
    expect(button).toHaveClass('w-full');
    expect(button).toHaveClass('cursor-pointer');
    expect(button).toHaveClass('select-none');
    expect(button).toHaveClass('items-center');
    expect(button).toHaveClass('gap-3');
    expect(button).toHaveClass('rounded-md');
  });

  it('renders text in a span element', () => {
    const mockOnClick = vi.fn();
    const { container } = render(<SidebarButton text="Span Text" icon={mockIcon} onClick={mockOnClick} />);

    const span = container.querySelector('span');
    expect(span).toBeInTheDocument();
    expect(span?.textContent).toBe('Span Text');
  });

  it('renders icon in a div element', () => {
    const mockOnClick = vi.fn();
    const { container } = render(<SidebarButton text="Icon Test" icon={mockIcon} onClick={mockOnClick} />);

    const iconDiv = container.querySelector('div');
    expect(iconDiv).toBeInTheDocument();
    expect(iconDiv).toContainElement(screen.getByTestId('test-icon'));
  });

  it('renders with different text values', () => {
    const mockOnClick = vi.fn();

    const { rerender } = render(<SidebarButton text="First" icon={mockIcon} onClick={mockOnClick} />);
    expect(screen.getByText('First')).toBeInTheDocument();

    rerender(<SidebarButton text="Second" icon={mockIcon} onClick={mockOnClick} />);
    expect(screen.getByText('Second')).toBeInTheDocument();
    expect(screen.queryByText('First')).not.toBeInTheDocument();
  });

  it('renders with different icons', () => {
    const mockOnClick = vi.fn();
    const icon1 = <svg data-testid="icon-1">Icon 1</svg>;
    const icon2 = <svg data-testid="icon-2">Icon 2</svg>;

    const { rerender } = render(<SidebarButton text="Test" icon={icon1} onClick={mockOnClick} />);
    expect(screen.getByTestId('icon-1')).toBeInTheDocument();

    rerender(<SidebarButton text="Test" icon={icon2} onClick={mockOnClick} />);
    expect(screen.getByTestId('icon-2')).toBeInTheDocument();
    expect(screen.queryByTestId('icon-1')).not.toBeInTheDocument();
  });

  it('handles empty text gracefully', () => {
    const mockOnClick = vi.fn();
    render(<SidebarButton text="" icon={mockIcon} onClick={mockOnClick} />);

    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();

    const span = button.querySelector('span');
    expect(span?.textContent).toBe('');
  });

  it('matches snapshot', () => {
    const mockOnClick = vi.fn();
    const { container } = render(<SidebarButton text="Snapshot Test" icon={mockIcon} onClick={mockOnClick} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
