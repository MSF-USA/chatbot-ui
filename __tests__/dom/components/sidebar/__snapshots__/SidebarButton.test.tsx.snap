// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SidebarButton > matches snapshot 1`] = `
<button
  class="flex w-full cursor-pointer select-none items-center gap-3 rounded-md py-3 px-3 text-[14px] leading-3 text-black dark:text-white transition-colors duration-200 hover:bg-gray-500/10"
>
  <div>
    <svg
      data-testid="test-icon"
    >
      Icon
    </svg>
  </div>
  <span>
    Snapshot Test
  </span>
</button>
`;
